### PR TITLE
insurance backend

### PR DESCRIPTION
**Pull request by isabellapedraza**

### CODE REVIEW

File: server/insurance/middleware.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+  next();
+  
+};
+
+/**
+ * Checks if a subscriber name in req.body is valid, that is, it matches the name regex
+ */
+const isValidSubscriberName = (req: Request, res: Response, next: NextFunction) => {
+    const nameRegex = /^(?!\s*$).+/i;
```
Comment (jrdegreeff) [inline server/insurance/middleware.ts]: what does this regex actually test for?

File: server/insurance/middleware.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    if (req.body.subscriber_name){
+      if (!nameRegex.test(req.body.subscriber_name)) {
+        res.status(400).json({
+        error: 'Subscriber name must be a nonempty alphanumeric string.'
+        });
+        return;
+    }
```
Comment (jrdegreeff) [inline server/insurance/middleware.ts]: [nit] indentation

**Review by jrdegreeff — CHANGES_REQUESTED** [review]
The additions look great to me! I added just a couple tiny comments inline.

The one major issue though is that in the last commit when you merged the changes from the other PR, you accidentally deleted all of the files from that PR. In other words if you were to merge this one it would completely delete the medical-contact code. What I would recommend is undoing that commit and doing the merge again without deleting those files. If you aren't sure how to undo a commit like that let me know and I can do it real quick for you.

File: server/insurance/middleware.ts (hunk 1 above)
Comment (isabellapedraza) [reply server/insurance/middleware.ts]: Just checks whether it's a non-empty space, will add in comment

**Review by jrdegreeff — APPROVED** [review]
lgtm!

**Pull request merged**